pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractReplicator.java

```diff
@@ -116,6 +118,13 @@ public String getRemoteCluster() {
     // This method needs to be synchronized with disconnects else if there is a disconnect followed by startProducer
     // the end result can be disconnect.
     public synchronized void startProducer() {
+        // This method comes from some actives call and may be call again after disconnect
+        // so here we will first mark isClosed is false
+        isClosed = false;
```

Comment:
Once the stat is set to `closed`, it can no longer be recovered to `no-closed`. If the replicator needs to keep working, it shouldn't be set to `closed`, right?

Reply:
Normally this is the case. But here, we can call the method `startProducer` again after call the method `disconnect`.
In fact, this class can be closed and opened again.

Reply:
> Normally this is the case. But here, we can call the method startProducer again after call the method disconnect.
In fact, this class can be closed and opened again.
> In fact, this class can be closed and opened again.

So we should only set the stat of Replicator to `closed` when it is exactly is closed, right?

Reply:
> Once the stat is set to `closed`, it can no longer be recovered to `no-closed`. If the replicator needs to keep working, it shouldn't be set to `closed`, right?

Normally this is the case. But here, we can call the method `startProducer` again after call the method `disconnect`.
In fact, this class can be closed and opened again.

Essentially, what we want to do here is to allow the retry thread (`checkTopicActiveAndRetryStartProducer`) to end normally when the replicator is closed.

Reply:
Once we add a stat, it must be meaningful, right?

Reply:
@codelipenghui @Technoboy- What do you think about this?

Reply:
> In fact, this class can be closed and opened again.

There is already a state `[Stopped, Starting, Started, Stopping]` is working on this case.

Reply:
There is currently no clear indication that this replicator will not be used again after it is `disconnect`.

Mabey it is not very reasonable to use the `close` status identifier here, but it can work normally.

I think it is also reasonable to use some other markers as judgment conditions.

Reply:
Can you provide a definition for this new state `isClosed`?  How is it different from `State.Stopped` when it is `true`?